galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/RestoreCPS.java

```diff
@@ -0,0 +1,294 @@
+     * @return
+     */
+    private String getPropertySuffix(String propertyName) throws ConfigurationPropertyStoreException {
+        return propSplit(propertyName, 1);
+    }
+    
+    /**
+     * <p>Splits a string into two parts: a prefix and a suffix.</p>
+     * <p>Prefix: Anything before the first dot "."</p>
+     * <p>Suffix: Anything after the first dot "."</p>
+     * 
+     * <p>Position specified as 0 or 1</p>
+     * 
+     * @param str
+     * @param position
+     * @return dashes
+     */
+    private String propSplit(String str, int position) throws ConfigurationPropertyStoreException {
+        
+        String[] kvp = str.split("\\.", 2);
+        if (kvp.length <= 1) {
+            throw new ConfigurationPropertyStoreException("Invalid Property Format: " + str);
+        }
+        return kvp[position];
+    }
+    
+    /**
+     * <p>Checks for property validity (whether there is a prefix and a suffix, separated by a dot ".").</p>
+     * 
+     * @param key
+     * @return boolean
+     */
+    private boolean isValidProperty(String key) {
+        /**
+         *  Regex matches (at least) three words (of one letter or more) separated by dots (".")
+         *  e.g. framework.foo.bar or framework.foo.bar.fizz.buzz
+         */
+        
+        Pattern pattern = Pattern.compile("^([a-zA-Z0-9]+\\.){2,}[a-zA-Z0-9]+$");
+        Matcher matcher = pattern.matcher(key);
+        boolean matchFound = matcher.find();
+        return matchFound;
+    }
+    
+    /**
+     * <p>Utility to fetch a string with a specified number of dashes.</p>
+     * <p>Useful for table output in console.</p>
+     * 
+     * @param num
+     * @return dashes
+     */
+    private String getDashes(int num) {
+        String dashes = new String(new char [num]).replace("\0", "-");
+        return dashes;
+    }
+    
+    /**
+     * <p>Retrieves the current/existing value for a specified property.</p>
+     * <p>This wrapper is required due to properties only being retrievable using both a prefix and a suffix.</p>
+     * 
+     * @param namespace
+     * @param key
+     * @return existingValue
+     * @throws ConfigurationPropertyStoreException 
+     */
+    private String getExistingValue(String namespace, String key) throws ConfigurationPropertyStoreException {
+     
+        String existingValue = 
+                namespaceCPS.get(namespace).getProperty(getPropertyPrefix(key), getPropertySuffix(key));
+                
+        return existingValue;
+        
+    }
+
+    /**
+     * <p>Tests namespace for restoration permission.</p>
+     * <ul>
+     *     <li>Returns true for valid namespace (restore permitted).</li>
+     *     <li>Returns false for invalid namespace (restore not permitted).</li>
+     * </ul>
+     * 
+     * @param namespace
+     * @return boolean
+     */ 
+    private boolean isNamespaceRestorePermitted(String namespace) {
+        List<String> forbiddenNamespaces = new ArrayList<String>();
```

Comment:
should need to run this code everytime the method is called,   convert to a HashSet,  move to the constructor

Reply:
Moved to the constructor.